src/main/java/mate/academy/mybookshop/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+package mate.academy.mybookshop.service;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import mate.academy.mybookshop.dto.order.CreateOrderRequestDto;
+import mate.academy.mybookshop.dto.order.OrderResponseDto;
+import mate.academy.mybookshop.dto.order.UpdateOrderRequestDto;
+import mate.academy.mybookshop.dto.orderitem.OrderItemResponseDto;
+import mate.academy.mybookshop.entity.CartItemEntity;
+import mate.academy.mybookshop.entity.OrderEntity;
+import mate.academy.mybookshop.entity.OrderItemEntity;
+import mate.academy.mybookshop.entity.UserEntity;
+import mate.academy.mybookshop.exception.EmptyCartException;
+import mate.academy.mybookshop.exception.EntityNotFoundException;
+import mate.academy.mybookshop.mapper.OrderItemMapper;
+import mate.academy.mybookshop.mapper.OrderMapper;
+import mate.academy.mybookshop.repository.CartItemRepository;
+import mate.academy.mybookshop.repository.OrderItemRepository;
+import mate.academy.mybookshop.repository.OrderRepository;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public OrderResponseDto createOrder(CreateOrderRequestDto requestDto, UserEntity user) {
```

Comment:
```suggestion
    @Override
    @Transactional
    public OrderResponseDto createOrder(CreateOrderRequestDto requestDto, UserEntity user) {
```
Because we have a creation of a set of entities and a deletion of a set of entities as a result of that operation and that operation should be atomic, and in general, in most of the situations you want your method to be transactional, unless you have absolutely huge or a small, single db call method

Reply:
added